src/app/EventScheduler.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * @brief
+ *   Define EventScheduler interface. Event scheduler is used by EventManagement to notify that events are ready to be scheduled.
+ *   Usually this is implemented by the Reporting Engine to find the proper ReadHandlers and deliver the events.
+ *
+ */
+#pragma once
+
+#include <app/ConcreteEventPath.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+
+class EventScheduler
```

Comment:
This should be called EventReporter or something like that.  It doesn't schedule events.  It reports events.  ScheduleEventDelivery is not really well-named either; it should be something like "NewEventGenerated()"; the fact that this does any scheduling of anything (or not, sometimes!) is implementation detail, not API....

We could have a general EventGenerationObserver interface, I guess, but I am not sure we need that.

I do think whatever interface we define here, the relevant method on it should be called NewEventGenerated or EventGenerated.

Reply:
Call it a EventReporter and change the method name.